### PR TITLE
Docs: Add example to the Indexer API doc

### PR DESCRIPTION
**Pull request by Sidnioulz**
## What I did

Added a doc example on the Indexer API, pointing to a Stackblitz that shows how to hack the Indexer API and a few other parts of SB to add external URLs to the sidebar.

This is useful because it shows how to connect arbitrary formats in the Indexer and in a Vite plugin that outputs CSF on each named export being indexed.

#### The changes in this PR are covered in the following automated tests:

_not relevant_

#### Manual testing

> [!CAUTION]
> Please note I couldn't live test the change. I couldn't manage to run frontpage and will require help to run it.

### Documentation

- [x] Add or update documentation reflecting your changes

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

## Greptile Summary

This PR enhances the Indexer API documentation by adding a new advanced example demonstrating how to create sidebar links from a collection of URLs.

- Added a detailed example in `docs/api/main-config/main-config-indexers.mdx` showcasing URL-based sidebar links
- Demonstrates integration of custom indexer, Vite plugin, and manager customization
- Includes a link to a StackBlitz demo for a complete working example
- Illustrates the flexibility of the Indexer API for processing non-standard story formats

### CODE REVIEW

File: docs/api/main-config/main-config-indexers.mdx
Comment (jonniebigodes) [inline docs/api/main-config/main-config-indexers.mdx:375]: @Sidnioulz, we have a bit of room for improvement here and we could go with the following:
````suggestion

<details>
  <summary>Adding sidebar links from a URL collection</summary>

  The indexer API is flexible enough to let you process arbitrary content, so long as your framework tooling can transform the exports in that content into actual stories it can run. This advanced example demonstrates how you can create a custom indexer to process a collection of URLs, extract the title and URL from each page, and render them as sidebar links in the UI. Implemented with Svelte, it can be adapted to any framework.

  Start by creating the URL collection file (i.e., `src/MyLinks.url.js`) with a list of URLs listed as named exports. The indexer will use the export name as the story title and the value as the unique identifier.

  ```js title="MyLinks.url.js"
  export default {};

  export const DesignTokens = 'https://www.designtokens.org/';
  export const CobaltUI = 'https://cobalt-ui.pages.dev/';
  export const MiseEnMode = 'https://mode.place/';
  export const IndexerAPI = 'https://github.com/storybookjs/storybook/discussions/23176';
  ```

  Adjust your Vite configuration file to include a custom plugin complementing the indexer. This will allow Storybook to process and import the URl collection file as stories.

  ```ts title="vite.config.ts"
  import * as acorn from 'acorn';
  import * as walk from 'acorn-walk';
  import { defineConfig, type Plugin } from 'vite';
  import { svelte } from '@sveltejs/vite-plugin-svelte';

  function StorybookUrlLinksPlugin(): Plugin {
    return {
      name: 'storybook-url-links',
      async transform(code: string, id: string) {
        if (id.endsWith('.url.js')) {
          const ast = acorn.parse(code, {
            ecmaVersion: 2020,
            sourceType: 'module',
          });

          const namedExports: string[] = [];
          let defaultExport = 'export default {};';

          walk.simple(ast, {
            // Extracts the named exports, those represent our stories, and for each of them, we'll return a valid Svelte component.
            ExportNamedDeclaration(node: acorn.ExportNamedDeclaration) {
              if (
                node.declaration &&
                node.declaration.type === 'VariableDeclaration'
              ) {
                node.declaration.declarations.forEach((declaration) => {
                  if ('name' in declaration.id) {
                    namedExports.push(declaration.id.name);
                  }
                });
              }
            },
            // Preserve our default export.
            ExportDefaultDeclaration(node: acorn.ExportDefaultDeclaration) {
              defaultExport = code.slice(node.start, node.end);
            },
          });

          return {
            code: `
              import RedirectBack from '../../.storybook/components/RedirectBack.svelte';
              ${namedExports
                .map(
                  (name) =>
                    `export const ${name} = () => new RedirectBack({ target: 'storybook-root'});`
                )
                .join('\n')}
              ${defaultExport}
            `,
            map: null,
          };
        }
      },
    };
  }

  export default defineConfig({
    plugins: [StorybookUrlLinksPlugin(), svelte()],
  })
  ```

  Update your Storybook configuration (i.e., `.storybook/main.js|ts`) to include the custom indexer.

  ```ts title=".storybook/main.js|ts"
  import type { StorybookConfig } from '@storybook/svelte-vite';
  import type { Indexer } from '@storybook/types';

  const urlIndexer: Indexer = {
    test: /\.url\.js$/,
    createIndex: async (fileName, { makeTitle }) => {
      const fileData = await import(fileName);

      return Object.entries(fileData)
        .filter(([key]) => key != 'default')
        .map(([name, url]) => {
          return {
            type: 'docs',
            importPath: fileName,
            exportName: name,
            title: makeTitle(name)
              .replace(/([a-z])([A-Z])/g, '$1 $2')
              .trim(),
            __id: `url--${name}--${encodeURIComponent(url as string)}`,
            tags: ['!autodocs', 'url']
          };
        });
    }
  };

  const config: StorybookConfig = {
    stories: ['../src/**/*.stories.@(js|ts|svelte)', '../src/**/*.url.js'],
    framework: {
      name: '@storybook/svelte-vite',
      options: {},
    },
    experimental_indexers: async (existingIndexers) => [urlIndexer, ...existingIndexers]
  };
  export default config;
  ```

  Add a Storybook UI configuration file (i.e., `.storybook/manager.js|ts`) to render the indexed URLs as sidebar links in the UI:

  ```ts title=".storybook/manager.ts"
  import { addons } from '@storybook/manager-api';

  import SidebarLabelWrapper from './components/SidebarLabelWrapper.tsx';

  addons.setConfig({
      sidebar: {
        renderLabel: (item) => SidebarLabelWrapper({ item }),
      },
  });
  ```

  This example's code and live demo are available on [StackBlitz](https://stackblitz.com/~/github.com/Sidnioulz/storybook-sidebar-urls).

</details>
````

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@Sidnioulz sorry for the delayed response on this pull request. I've checked and left one small item for you to look into when you can.

Let me know and we'll go from there.

Have a great day.

Stay safe

File: docs/api/main-config/main-config-indexers.mdx
Comment (Sidnioulz) [reply docs/api/main-config/main-config-indexers.mdx:375]: Thanks for the iteration on copy, @jonniebigodes :pray:

There's a tiny casing typo in your updated version, so I've taken the liberty to edit your suggestion before merging it.

**Review by jonniebigodes — APPROVED** [review]
@Sidnioulz, thank you for addressing the feedback and fixing the minor casing typo. Appreciate it 🙏 !  I've checked, and we're good to go on this end. I will gladly merge this once the checklist clears.

One small item for you to look into when you can. I was wondering if you have a bit of time to adjust the Stackblitz so that the URL links are easily readable for anyone who checks it, as you can see below:
Content-Indexers-UI-issue


Have a great day.

Stay safe

**Comment by Sidnioulz** [comment]
> @Sidnioulz, thank you for addressing the feedback and fixing the minor casing typo. Appreciate it 🙏 ! I've checked, and we're good to go on this end. I will gladly merge this once the checklist clears.
> 
> One small item for you to look into when you can. I was wondering if you have a bit of time to adjust the Stackblitz so that the URL links are easily readable for anyone who checks it, as you can see below: Content-Indexers-UI-issue
> 
> Have a great day.
> 
> Stay safe

Thanks for letting me know! I've switched to using `light-dark` so light colours have a default that works now. Didn't want to import emotion into the showcase as it's kind of beside the point, but I will in the future if others find other issues with this code.

**Pull request merged**
